src/torchmetrics/functional/regression/weighted_pearson.py

```diff
@@ -0,0 +1,176 @@
+
+
+def _weighted_pearson_corrcoef_update(
+    preds: Tensor,
+    target: Tensor,
+    weights: Tensor,
+    mean_x: Tensor,
+    mean_y: Tensor,
+    var_x: Tensor,
+    var_y: Tensor,
+    cov_xy: Tensor,
+    weights_prior: Tensor,
+    num_outputs: int,
+) -> tuple[Tensor, Tensor, Tensor, Tensor, Tensor, Tensor]:
+    """Update and returns variables required to compute weighted Pearson Correlation Coefficient.
+
+    Check for same shape of input tensors.
+
+    Updates are based on `Algorithms for calculating variance`_. Specifically, `online weighted variance`_ and
+    `online weighted covariance`_.
+
+    Variance intentionally not divided by sum of weights in `update` step as it is computed as necessary in
+    the `compute` step.
+
+    Args:
+        preds: estimated scores
+        target: ground truth scores
+        weights: weight associated with scores
+        mean_x: current mean estimate of x tensor
+        mean_y: current mean estimate of y tensor
+        var_x: current variance estimate of x tensor
+        var_y: current variance estimate of y tensor
+        cov_xy: current covariance estimate between x and y tensor
+        weights_prior: current sum of weights
+        num_outputs: number of outputs in multioutput setting
+
+    """
+    # Data checking
+    _check_same_shape(preds, target)
+    _check_data_shape_to_num_outputs(preds, target, num_outputs)
+    _check_data_shape_to_weights(preds, weights)
+
+    weights_sum = weights.sum()
+
+    if weights_prior > 0:  # True if prior observations exist
+        mx_new = (weights_prior * mean_x + (weights * preds).sum(0)) / (weights_prior + weights_sum)
+        my_new = (weights_prior * mean_y + (weights * target).sum(0)) / (weights_prior + weights_sum)
+
+        var_x += (weights * (preds - mx_new) * (preds - mean_x)).sum(0)
+        var_y += (weights * (target - my_new) * (target - mean_y)).sum(0)
+    else:
+        mx_new = ((weights * preds).sum(0) / weights_sum).to(mean_x.dtype)
+        my_new = ((weights * target).sum(0) / weights_sum).to(mean_y.dtype)
+
+        var_x = (weights * (preds - mx_new) ** 2).sum(0)
+        var_y = (weights * (target - my_new) ** 2).sum(0)
+
+    cov_xy += (weights * (preds - mx_new) * (target - mean_y)).sum(0)
+
+    return mx_new, my_new, var_x, var_y, cov_xy, weights_prior + weights_sum
+
+
+def _weighted_pearson_corrcoef_compute(
+    var_x: Tensor,
+    var_y: Tensor,
+    cov_xy: Tensor,
+    weights_sum: Tensor,
+) -> Tensor:
+    """Compute the final weighted Pearson correlation based on accumulated statistics.
+
+    Args:
+        var_x: variance estimate of x tensor
+        var_y: variance estimate of y tensor
+        cov_xy: covariance estimate between x and y tensor
+        weights_sum: sum of weights
+
+    """
+    # prevent overwrite the inputs
+    var_x = var_x / (weights_sum - 1)
+    var_y = var_y / (weights_sum - 1)
+    cov_xy = cov_xy / (weights_sum - 1)
```

Comment:
are we sure that the `weights_sum - 1` normalization is needed here? the reference implementation does not have this normalization

Reply:
You are right, it's just `weights_sum` normalization here.